lib/phoenix/channel/server.ex

```diff
@@ -147,9 +147,9 @@ defmodule Phoenix.Channel.Server do
   def reply(pid, ref, topic, {status, payload}, serializer)
       when is_binary(topic) and is_map(payload) do
 
-    send pid, serializer.encode!(
+    send pid, {:channel_push, serializer.encode!(
```

Comment:
Yes, these are the messages sent out by the channel process to its socket, instructing it to push message/reply to the client. So the socket driver needs to handle these messages in a particular way, by producing the outgoing message. See [here](https://github.com/sasa1977/phoenix/blob/transport-behaviour/lib/phoenix/channel/driver.ex#L96-L98).